lib/auth.js

```diff
@@ -48,6 +55,66 @@ Auth.prototype.bearer = function (bearer, sendImmediately) {
   }
 }
 
+
+Auth.prototype.negotiate = function(request, authHeader, sendImmediately) {
+  var self = this
+
+  if (kerberos === undefined) {
+    // Nothing to do if we don't have Kerberos
+    // or a current credential cache
+    return
+  }
+
+  self.hasAuth = true
+
+  if (!sendImmediately) {
```

Comment:
Ok, just don't forget to execute the commands from my first comment https://github.com/request/request/pull/1966#issuecomment-164790235 you may not need all of them though.

Also I just added the:

```bash
# obtain a ticket-granting ticket using the keytab
$ kinit -k -t /etc/krb5.keytab HTTP/localhost
```

Not sure how is this supposed to work in the real world, I mean I can't find such method in the `kerberos` module.